crates/transaction-pool/src/pool/mod.rs

```diff
@@ -553,6 +567,26 @@ where
         });
     }
 
+    /// Notify all listeners about a blob sidecar for a newly inserted blob (eip4844) transaction.
+    fn on_new_blob_sidecar(&self, sidecar: BlobTransactionSidecar) {
```

Comment:
this should take by ref

Reply:
done, although I have a question here..

I have now changed it to take a reference for both `tx_hash` and `sidecar` as such:
`self.on_new_blob_sidecar(&hash, &sidecar);` 

and inside the function I have:
```
 let new_blob_event = NewBlobSidecar { tx_hash: *tx_hash, sidecar: *sidecar };
        sidecar_listeners.retain_mut(|listener| match listener.sender.try_send(new_blob_event)
```

I'm confused as to how `*sidecar` there is different from `sidecar.clone()`(with `&BlobTransactionSidecar`) is different from passing in `sidecar.clone()` with (`sidecar: BlobTransactionSideCar`)

The way I understand the deref * on a `Clone` type is it effectively allocates memory on the heap for the value behind the reference.

If my understanding is correct (it may not be) - am I really getting away from the `clone()` by doing what I've done above? (I think not)

The way to solve this I think is to modify `NewBlobSidecar` as such:
```
pub struct NewBlobSidecar<'a> {
 /// The blob transaction hash.
pub tx_hash: &TxHash,
/// the blob transaction sidecar.
pub sidecar: &'a BlobTransactionSidecar,
```

add the references to `tx_hash` and `sidecar` to `NewBlobSidecar` along with a non-static lifetime (?)